tests/Unit/Controllers/CourseControllerTest.php

```diff
@@ -0,0 +1,214 @@
+<?php
+
+namespace Tests\Unit\Controllers;
+
+use App\Http\Controllers\Auth\Admin\CourseAdminController;
+use App\Models\Course;
+use Tests\TestCase;
+use Mockery as m;
+use Illuminate\Database\Connection;
+use Illuminate\Http\Request;
+use Symfony\Component\HttpFoundation\ParameterBag;
+use Illuminate\Database\QueryException;
+use Illuminate\Http\RedirectResponse;
+
+class CourseControllerTest extends TestCase
+{
+    protected $db;
+    protected $courseMock;
+
+    public function setUp(): void
+    {
+        $this->afterApplicationCreated(function () {
+            $this->db = m::mock(
+                Connection::class . '[select,update,insert,delete]',
```

Comment:
function này dùng để làm gì?

Reply:
vì nó nằm trong setUp() nên cứ mỗi khi pass xong 1 function thì hàm setUp() sẽ khởi tạo lại các variable bên trong nó,
còn afterApplicationCreated() sẽ khởi tạo một cuộc gọi để được chạy sau khi ứng dụng được tạo, trong đó mình sẽ thiết lập các variable toàn cục để sử dụng.
*Nếu em giải thích không đúng chị giải thích giúp em với ạ.